src/test/compile-fail/union-field-privacy.rs

```diff
@@ -0,0 +1,21 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
```

Comment:
For all these tests with names like `union-foo`, let's make a `union/` directory. (Also, I think we should just go ahead and move them all to `ui` tests, per the logic in [this thread](https://internals.rust-lang.org/t/testing-the-compiler/3836).)